logfire/_internal/config.py

```diff
@@ -174,6 +177,25 @@ class MetricsOptions:
     """Sequence of metric readers to be used in addition to the default which exports metrics to Logfire's API."""
 
 
+@dataclass
+class CodeSource:
+    """Settings for the source code of the project."""
+
+    repository: str
+    """The repository URL for the code e.g. https://github.com/pydantic/logfire"""
+
+    revision: str
+    """The git revision of the code e.g. branch name, commit hash, tag name etc."""
+
+    root_path: str
+    """The root path for the source code in the repository.
+
+    Example:
+        If the `code.filename` is `/path/to/project/src/logfire/main.py` and the `root_path` is `src/`, the URL
```

Comment:
I'm planning to document this feature btw... 

This PR is just the first step. After the feature is on the frontend, I'll add to the docs.

Reply:
> Logfire spans automatically have a `code.filepath` attribute indicating where they were created. That path is relative to the current working directory of the Python process. The `root_path` setting is combined with the `code.filepath` attribute to construct a link, so `root_path` should be a path starting from the root of the repository that corresponds to the current working directory of the Python process.

> For example, suppose that your repository contains `a/b/c/main.py`, the folder `a/b/` is copied into the `/docker/root/` folder of your docker container, and within the container the command `python ./b/c/main.py` is run from within the `/docker/root/a/` directory. Then `code.filepath` will be `b/c/main.py` for spans created in that file, and the `root_path` should be set to `a` so that the final link is `a/b/c/main.py`.